File: readability/readability.c
#include <cs50.h>
#include <stdio.h>
#include <string.h>
#include <ctype.h>
#include <math.h>


// function to count characters in string
int count_letters(string a);

// function to count words in string
int count_words(string a);

// function to count sentences
int count_sentences(string a);

int main(void)

{
    string s = get_string("Text:  ");

    int letters = count_letters(s);

    int words = count_words(s);

    int sentences = count_sentences(s);

    float L = (((float) letters / (float) words) * 100);

    float S = (((float)sentences / (float) words) * 100);

    int index = round(0.0588 * L - 0.296 * S - 15.8);

    if (index >= 16)
    {
        printf("Grade 16+\n");
    }
    else if (index < 1)
    {
        printf("Before Grade 1\n");
    }
    else
    {
        printf("Grade %i\n", index);
    }

}


















// my function for counting letters


int count_letters(string a)
{
    int letters = 0;

    for (int i = 0, n = strlen(a); i < n ; i++)
    {
        if (isalpha(a[i]))
        {
            letters++;
        }
        // alternative:  isalpha(a[i]) ? letters++ : letters ;
    }
    return letters;
}










// function to count words in string
int count_words(string a)
{
    int words = 0;

    if ((isblank(a[0]) != true))
    {
        words++;
    }

    for (int i = 0, n = strlen(a); i < n ; i++)
    {
        if (a[i] == ' ' && (isblank(a[i + 1]) != true))
        {
            words++;
        }
    }
    return words;
}









// my function for counting sentences


int count_sentences(string a)
{
    int sentences = 0;

    for (int i = 0, n = strlen(a); i < n ; i++)
    {
        if (a[i] == '.' || a[i] == '?' || a[i] == '!')
        {
            sentences++;
        }
    }
    return sentences;
}

File: pset6/readability/readability.py
from cs50 import get_string
from math import floor


def main():

    s = get_string("Text: ")

    letters = count_letters(s)

    words = count_words(s)

    sentences = count_sentences(s)

    L = (letters / words) * 100

    S = sentences / words * 100

    index = 0.0588 * L - 0.296 * S - 15.8

    if index >= 16:
        print("Grade 16+")
    elif index < 1:
        print("Before Grade 1")
    else:
        print("Grade", round(index))


def count_letters(str):
    letters = 0

    for i in range(0, len(str), 1):
        if str[i].isalpha():
            letters += 1
    return letters


def count_words(str):

    words = 0
    if str[0].isspace() != True:
        words += 1
    for i in range(0, len(str), 1):
        if (str[i]) == ' ' and (str[i+1].isspace() != True):
            words += 1
    return words


def count_sentences(str):

    sentences = 0

    for i in range(0, len(str), 1):
        if str[i] == '.' or str[i] == '?' or str[i] == '!':
            sentences += 1
    return sentences


main()